driver-core/src/main/java/com/datastax/driver/core/Frame.java

```diff
@@ -205,10 +200,17 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> ou
             int version = buffer.getByte(0);
             // version first bit is the "direction" of the frame (request or response)
             version = version & 0x7F;
+
+            DecoderForStreamIdSize decoder = decoderForStreamIdSize.get();
```

Comment:
Needed because we can't be sure the protocol version the server is using until we actually get something from it.  Although I'm not sure an `AtomicReference` is necessary here, i'd expect `decode` will only called one at a time and from the same i/o thread, wdyt?  In general could there be a better approach than this?